ground/src/main/java/com/google/android/ground/persistence/local/room/converter/LoiPropertiesMapConverter.kt

```diff
@@ -0,0 +1,45 @@
+package com.google.android.ground.persistence.local.room.converter
+
+import androidx.room.TypeConverter
+import com.google.common.reflect.TypeToken
+import com.google.gson.Gson
+import com.google.gson.JsonSyntaxException
+import kotlinx.collections.immutable.toPersistentMap
+import org.json.JSONException
+import org.json.JSONObject
+import timber.log.Timber
+
+
+object LoiPropertiesMapConverter {
+  @TypeConverter
+  fun toString(properties: Map<String, Any>?): String? {
+    if (properties ==null) {
+      return null
+    }
+    var jsonString: String? = null
+    try {
+      jsonString =  JSONObject(properties).toString()
+    } catch (e: JSONException) {
+      Timber.e(e, "Error building JSON")
+    }
+    return jsonString
+  }
+
+  @TypeConverter
+  fun fromString(jsonString: String?): Map<String, Any> {
+    if (jsonString == null) {
+      return mapOf()
+    }
+    var map = mutableMapOf<String, Any>()
+    try {
+      val type = object: TypeToken<Map<String, Any>>(){}.type
+      map = Gson().fromJson(
```

Comment:
@JSunde introduce proto (de)serialization for local storage. Can we use that instead of packing/unpacking Json ourselves?

Reply:
Is this referring to JsonObjectTypeConverter?

Reply:
It's being done in `GeometryWrapperTypeConverter` using`ProtoBuf.encodeToByteArray()`. Can we use the same here?

Reply:
Hmmm so using ProtoBuf.encodeToByteArray() throws a Serialization Exception because kotlin doesn't have a serializer for the Any class http://screen/H4ovCcLxGR3MvZA

This medium article suggests either creating a custom serializer, or defining a stricter type, or using Gson. https://medium.com/@kerem_balic/serializer-for-class-any-not-found-how-do-we-fix-it-in-kotlin-cc196d92b7b3

What are your thoughts?

Reply:
Yikes, sorry I hadn't thought of that. Since we're adding properties to the model, perhaps now would be a good time to add stricter typing? Afaik there are only two possible types for properties values: `String` and `Number`. 

Assume the proto serializer can handle generics, can we define an interface `PropertyValue<T>` with one member `T value`, which is implemented by `NumberValue: PropertyValue<Number>` and `StringValue: PropertyValue<String>`?

Reply:
We chatted. We can make this change in a follow-up PR!